File: MusicDemoOC/MusicDemo/View/LrcLabel.m
//
//  LrcLabel.m
//  MusicDemo
//
//  Created by 沈凯 on 2017/11/9.
//  Copyright © 2017年 Ssky. All rights reserved.
//

#import "LrcLabel.h"

@implementation LrcLabel

- (void)setProgress:(CGFloat)progress
{
    _progress = progress;
    
    [self setNeedsDisplay];
}
- (void)drawRect:(CGRect)rect
{
    [super drawRect:rect];
    
    CGRect drawRect = CGRectMake(0, 0, self.bounds.size.width * self.progress, self.bounds.size.height);
    [[UIColor greenColor] set];
    UIRectFillUsingBlendMode(drawRect, kCGBlendModeSourceIn);
}
@end

File: MusicDemoSwift/MusicDemoSwift/View/LrcLabel.swift
//
//  LrcLabel.swift
//  MusicDemoSwift
//
//  Created by 沈凯 on 2017/11/20.
//  Copyright © 2017年 Ssky. All rights reserved.
//

import UIKit

class LrcLabel: UILabel {
//    当前语句的进度
    var progress: CGFloat {
        set{
            setNeedsDisplay()
        }
        get{
            return self.progress
        }
    }
    override func draw(_ rect: CGRect) {
        super.draw(rect)
        let drawRect = CGRect(x: 0, y: 0, width: bounds.size.width * progress, height: bounds.size.height)
        UIColor.green.set()
        UIRectFillUsingBlendMode(drawRect, .sourceIn)
    }
}
